memorious/operations/aleph.py

```diff
@@ -132,6 +136,57 @@ def aleph_folder(context, data):
             backoff(ae, try_number)
 
 
+def aleph_entity(context, data):
+    api = get_api(context)
+    if api is None:
+        return
+    collection_id = get_collection_id(context, api)
+    content_hash = data.get("content_hash")
+    source_url = data.get("source_url", data.get("url"))
+    foreign_id = data.get("foreign_id", data.get("request_id", source_url))
+    # Fetch document id from cache
+    document_id = context.get_tag(make_key(collection_id, foreign_id, content_hash))
+    if document_id:
+        context.log.info("Skip aleph upload: %s", foreign_id)
+        data["aleph_id"] = document_id
+        context.emit(data=data, optional=True)
+        return
+    else:
+        document_id = uuid.uuid4().hex
```

Comment:
This is a bit confusing to me because we are conflating `document` and `entity`. An entity shouldn't be expected to have a `content_hash` or a `document_id`. May be we should expect an unique `entity_id` instead and use that to generate the caching key.

Reply:
..so the challenge here is that an article is a type of entity but also a type of document.